cura/Snapshot.py

```diff
@@ -182,11 +193,19 @@ def snapshot(width = 300, height = 300):
             preview_pass.setCamera(camera)
             preview_pass.render()
             pixel_output = preview_pass.getOutput()
-            try:
-                min_x, max_x, min_y, max_y = Snapshot.getImageBoundaries(pixel_output)
-            except (ValueError, AttributeError):
-                Logger.logException("w", "Failed to crop the snapshot!")
+            if Snapshot._prereadSnapshot(pixel_output):
+                try:
+                    min_x, max_x, min_y, max_y = Snapshot.getImageBoundaries(pixel_output)
+                except (ValueError, AttributeError) as e:
+                    Logger.logException("w", f"Failed to crop the snapshot! {e}")
+                    return None
+            elif number_of_attempts == 0:
+                Logger.warning( f"Failed to crop the snapshot even after 10 attempts!")
                 return None
+            else:
+                number_of_attempts = number_of_attempts - 1
+                Logger.info("Trying to get the snapshot again.")
+                return Snapshot.snapshot(width, height, number_of_attempts)
```

Comment:
I personally would have written this block as the following as I think the `Snapshot._prereadSnapshot(pixel_output)` call is already implicitly incorporated in the `Snapshot.getImageBoundaries(pixel_output)` call
```python
            try:
                min_x, max_x, min_y, max_y = Snapshot.getImageBoundaries(pixel_output)
            except (ValueError, AttributeError) as e:
                if number_of_attempts == 0:
                    Logger.warning( f"Failed to crop the snapshot even after 10 attempts!")
                    return None
                else:
                    number_of_attempts = number_of_attempts - 1
                    Logger.info("Trying to get the snapshot again.")
                    return Snapshot.snapshot(width, height, number_of_attempts)
```

Reply:
oh yes! Then preread function can go (y)

Reply:
[c488164](https://github.com/Ultimaker/Cura/pull/18549/commits/c4881641c3c50add2bfe0b991bc264c82168fb25)